pulpcore/cli/deb/content.py

```diff
@@ -0,0 +1,321 @@
+    ),
+    pulp_option(
+        "--source",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--tag",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--version",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--component",
+        allowed_with_contexts=(
+            PulpDebInstallerFileIndexContext,
+            PulpDebPackageIndexContext,
+            PulpDebReleaseComponentContext,
+        ),
+    ),
+    pulp_option(
+        "--release-component", allowed_with_contexts=(PulpDebPackageReleaseComponentContext,)
+    ),
+]
+lookup_options = [
+    href_option,
+    pulp_option(
+        "--relative-path",
+        callback=_relative_path_callback,
+        expose_value=False,
+        allowed_with_contexts=(
+            PulpDebGenericContentContext,
+            PulpDebInstallerFileIndexContext,
+            PulpDebPackageIndexContext,
+            PulpDebPackageContext,
+            PulpDebReleaseFileContext,
+        ),
+    ),
+    pulp_option(
+        "--sha256",
+        callback=_sha256_callback,
+        expose_value=False,
+        allowed_with_contexts=(
+            PulpDebGenericContentContext,
+            PulpDebInstallerFileIndexContext,
+            PulpDebPackageIndexContext,
+            PulpDebPackageContext,
+            PulpDebReleaseFileContext,
+        ),
+    ),
+]
+
+content.add_command(list_command(decorators=list_options))
+content.add_command(show_command(decorators=lookup_options))
+# create assumes "there exists an Artifact..."
+# create is defined for package and expects the sha256 of the artifact.
+create_options = [
+    pulp_option(
+        "--relative-path",
+        allowed_with_contexts=(PulpDebPackageContext,),
+    ),
+    pulp_option(
+        "--sha256",
+        "artifact",
+        required=True,
+        help=_("Digest of the artifact to use"),
+        callback=_sha256_artifact_callback,
+        allowed_with_contexts=(PulpDebPackageContext,),
+    ),
+    repository_option,
+]
+content.add_command(
+    create_command(
+        decorators=create_options,
+        allowed_with_contexts=(PulpDebPackageContext,),
+    )
+)
+
+
+# upload takes a file-argument and creates the entity from it.
+# upload currently only works for packages.
+@content.command(allowed_with_contexts=(PulpDebPackageContext,))
+@chunk_size_option
+@pulp_option(
+    "--relative-path",
+    help=_("Relative path within a distribution of the entity"),
+    allowed_with_contexts=(PulpDebPackageContext,),
+)
+@pulp_option(
+    "--file",
+    type=click.File("rb"),
+    required=True,
+    help=_("An DEB binary"),
+    allowed_with_contexts=(PulpDebPackageContext,),
+)
+@repository_option
```

Comment:
This option does not seem to work for me:
```
${PULP_CLI_CMD} deb content upload --repository ${REPOSITORY_HREF} --file ../frigg_1.0_ppc64.deb
Error: The type 'deb:apt' is not valid for the repository option.
```

Reply:
Works fine for me.

```
$ oci-env pulp deb content upload --repository pulp --file /src/pulp_deb/pulp_deb/tests/functional/data/packages/frigg_1.0_ppc64.deb
2023-08-17 09:23:58,298 - INFO - utils.py:37 - USING /home/hstct/dev/oci/oci_env FOR OCI_ENV_PATH BASED ON GIT CMD OUTPUT
2023-08-17 09:23:58,299 - INFO - utils.py:37 - USING /home/hstct/dev/oci/oci_env FOR OCI_ENV_PATH BASED ON GIT CMD OUTPUT
2023-08-17 09:23:58,300 - INFO - utils.py:37 - USING /home/hstct/dev/oci/oci_env FOR OCI_ENV_PATH BASED ON GIT CMD OUTPUT
Started background task /pulp/api/v3/tasks/018a0261-31d5-795c-8555-81cafc2fafbf/
.Done.
{
  "pulp_href": "/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/",
  "pulp_created": "2023-08-17T07:23:58.893158Z",
  "number": 2,
  "repository": "/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/",
  "base_version": null,
  "content_summary": {
    "added": {
      "deb.package": {
        "count": 1,
        "href": "/pulp/api/v3/content/deb/packages/?repository_version_added=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      },
      "deb.package_release_component": {
        "count": 1,
        "href": "/pulp/api/v3/content/deb/package_release_components/?repository_version_added=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      },
      "deb.release_architecture": {
        "count": 1,
        "href": "/pulp/api/v3/content/deb/release_architectures/?repository_version_added=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      },
      "deb.release_component": {
        "count": 1,
        "href": "/pulp/api/v3/content/deb/release_components/?repository_version_added=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      }
    },
    "removed": {
      "deb.package": {
        "count": 1,
        "href": "/pulp/api/v3/content/deb/packages/?repository_version_removed=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      },
      "deb.package_index": {
        "count": 4,
        "href": "/pulp/api/v3/content/deb/package_indices/?repository_version_removed=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      },
      "deb.release_file": {
        "count": 1,
        "href": "/pulp/api/v3/content/deb/release_files/?repository_version_removed=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      }
    },
    "present": {
      "deb.package": {
        "count": 4,
        "href": "/pulp/api/v3/content/deb/packages/?repository_version=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      },
      "deb.package_release_component": {
        "count": 5,
        "href": "/pulp/api/v3/content/deb/package_release_components/?repository_version=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      },
      "deb.release": {
        "count": 1,
        "href": "/pulp/api/v3/content/deb/releases/?repository_version=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      },
      "deb.release_architecture": {
        "count": 3,
        "href": "/pulp/api/v3/content/deb/release_architectures/?repository_version=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      },
      "deb.release_component": {
        "count": 3,
        "href": "/pulp/api/v3/content/deb/release_components/?repository_version=/pulp/api/v3/repositories/deb/apt/0189feb0-f9e3-7050-b2dd-1faa005381ab/versions/2/"
      }
    }
  }
}
```

Reply:
I assume however we don't want the `--relative-path` option here as well?

Reply:
I re-tested this with a different package and it worked. :shrug: